neelance/errors/errors.go

```diff
@@ -29,10 +30,11 @@ func Errorf(format string, a ...interface{}) *QueryError {
 
 // WithMessagef is the same as Errorf, except it will store the err inside
 // the ResolverError field.
-func WithMessagef(err error, format string, a ...interface{}) *QueryError {
+func WithMessagef(err error, extra interface{}, format string, a ...interface{}) *QueryError {
```

Comment:
This is the worst kind of BC break. 

`WithMessagef(err, "hello %s", name)` will continue to compile but the format string will dumped into extra and the name into the format string.